test/tests/functional/pbs_qmgr.py

```diff
@@ -0,0 +1,126 @@
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import os
+
+from tests.functional import *
+from ptl.lib.pbs_ifl_mock import *
+
+
+class TestQmgr(TestFunctional):
+    """
+    Test suite for PBSPro's qmgr command
+    """
+
+    def __check_whitespace_prefix(self, line):
+        """
+        Check whether the whitespace prefix for the line specified is correct
+
+        :param line: the line to check
+        :type line: String
+        """
+        if line is None:
+            return
+
+        if line[0] == " ":
+            # Spaces are prefix for new attribute lines
+            # Make sure that this is a new attribute line
+            self.assertTrue("=" in line)
+        elif line[0] == "\t":
+            # Tabs are prefix for line extensions
+            # Make sure that this is a line extension
+            self.assertTrue("=" not in line)
+
+    def test_listcmd_whitespaces(self):
+        """
+        Check that the prefix for new attributes listed out by qmgr list
+        are spaces and that for line extensions is a tab
+        """
+        fd, fn = self.du.mkstemp()
+        node_prefix = "vn"
+        nodename = node_prefix + "[0]"
+        vndef_file = None
+        try:
+            # Check 1: New attributes are prefixed with spaces and not tabs
+            # Execute qmgr -c 'list sched' and store output in a temp file
+            if self.du.is_localhost(self.server.hostname) is True:
+                qmgr_cmd = ["qmgr", "-c", "list sched"]
+            else:
+                qmgr_cmd = ["qmgr", "-c", "\'list sched\'"]
+            with os.fdopen(fd, "w+") as tempfd:
+                ret = self.du.run_cmd(self.server.hostname, qmgr_cmd,
+                                      stdout=tempfd)
+
+                self.assertTrue(ret['rc'] == 0)
+                for line in tempfd:
```

Comment:
ret['out'] doesn't return the 'pristine' output of the command, it returns a list of the contents, and because I'm looking for something as subtle as whitespaces in the output, I wanted to look at the pristine, un-process output. That's why I think it's better to just write it out to a file and read it back in. What do you think?